### PR TITLE
Add IP.{Interface,Link}LocalMulticast methods

### PR DESCRIPTION
**Pull request by tklauser**
Corresponding to the respective standard library `net.IP` methods: [`net.IP.IsInterfaceLocalMulticast`](https://golang.org/pkg/net/#IP.IsInterfaceLocalMulticast) and [`net.IP.IsLinkLocalMulticast`](https://golang.org/pkg/net/#IP.IsLinkLocalMulticast).

### CODE REVIEW

File: netaddr.go [hunk 1]
```diff
@@ -564,6 +564,27 @@ func (ip IP) IsMulticast() bool {
 	return false // zero value
 }
 
+// IsInterfaceLocalMulticast reports whether ip is an interface-local multicast
```
Comment (mdlayher) [inline netaddr.go:567]: Seems worth noting that this only applies to IPv6 addresses in the doc comment.

**Review by mdlayher — APPROVED** [review]
LGTM with a docs nit.

File: netaddr.go (hunk 1 above)
Comment (tklauser) [reply netaddr.go]: Done, thanks.

**Review by mdlayher — APPROVED** [review]
LGTM thanks

**Pull request merged**